packages/frontend/src/components/common/ShareSpaceModal/ShareSpaceUserList.tsx

```diff
@@ -529,7 +529,27 @@ export const ShareSpaceUserList: FC<ShareSpaceUserListProps> = ({
     );
 
     const accessByType = useMemo<SpaceAccessByType>(() => {
-        return space.access.reduce<SpaceAccessByType>(
+        const roleOrder = {
+            // higher roles have higher numbers
+            [SpaceMemberRole.VIEWER]: 1,
+            [SpaceMemberRole.EDITOR]: 2,
+            [SpaceMemberRole.ADMIN]: 3,
+        };
+
+        // function to get the highest role for a user
+        const getHighestAccess = (
+            existing: SpaceShare,
+            current: SpaceShare,
+        ) => {
+            if (existing.hasDirectAccess) return existing; // direct access takes priority
+            const existingRoleNumber = roleOrder[existing.role];
+            const currentRoleNumber = roleOrder[current.role];
+            return currentRoleNumber > existingRoleNumber ? current : existing;
+        };
+
+        const deduplicatedUsersMap = new Map<string, SpaceShare>();
+
+        const result = space.access.reduce<SpaceAccessByType>(
```

Comment:
We can update the reducer type and remove unused logic (push to direct array) and avoid a separate variable to track the dedupe user map.

Example new type 
```
reduce<{
    project: SpaceShare[];
    organisation: SpaceShare[];
    directMap: Map<string, SpaceShare>;
}>
```